Can't set Version or Priority value in plugin due to issue in plugin server

Currently, there seems to be an issue when the plugin sets the `Version` or `Priority` value in the plugin code. Example

```
var Version = "0.1.0"
```

When using the plugin, a big error is thrown in Kong

```
panic: interface conversion: plugin.Symbol is *string, not string

goroutine 1 [running]:
main.(*PluginServer).GetPluginInfo(0xc0000120a0, 0x7ffeefbfef20, 0x15, 0xc0000d3e40, 0x0, 0x0)
        /Users/karuppiahn/oss/github.com/kong/go-pluginserver/pluginserver.go:259 +0x806
main.dumpInfo()
        /Users/karuppiahn/oss/github.com/kong/go-pluginserver/main.go:48 +0x8e
main.main()
        /Users/karuppiahn/oss/github.com/kong/go-pluginserver/main.go:114 +0x73
2020/01/15 13:49:39 [error] 23230#0: init_by_lua error: /usr/local/share/lua/5.1/MessagePack.lua:813: missing bytes
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/MessagePack.lua:813: in function 'underflow'
        /usr/local/share/lua/5.1/MessagePack.lua:529: in function 'unpack_cursor'
        /usr/local/share/lua/5.1/MessagePack.lua:843: in function 'unpack'
        /usr/local/share/lua/5.1/kong/db/dao/plugins/go.lua:383: in function 'get_plugin_info'
        /usr/local/share/lua/5.1/kong/db/dao/plugins/go.lua:392: in function 'get_plugin'
        /usr/local/share/lua/5.1/kong/db/dao/plugins/go.lua:426: in function 'load_plugin'
        /usr/local/share/lua/5.1/kong/db/dao/plugins.lua:151: in function 'load_plugin_handler'
        /usr/local/share/lua/5.1/kong/db/dao/plugins.lua:227: in function 'load_plugin'
        /usr/local/share/lua/5.1/kong/db/dao/plugins.lua:275: in function 'load_plugin_schemas'
        /usr/local/share/lua/5.1/kong/init.lua:425: in function 'init'
        init_by_lua:3: in main chunk
nginx: [error] init_by_lua error: /usr/local/share/lua/5.1/MessagePack.lua:813: missing bytes
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/MessagePack.lua:813: in function 'underflow'
        /usr/local/share/lua/5.1/MessagePack.lua:529: in function 'unpack_cursor'
        /usr/local/share/lua/5.1/MessagePack.lua:843: in function 'unpack'
        /usr/local/share/lua/5.1/kong/db/dao/plugins/go.lua:383: in function 'get_plugin_info'
        /usr/local/share/lua/5.1/kong/db/dao/plugins/go.lua:392: in function 'get_plugin'
        /usr/local/share/lua/5.1/kong/db/dao/plugins/go.lua:426: in function 'load_plugin'
        /usr/local/share/lua/5.1/kong/db/dao/plugins.lua:151: in function 'load_plugin_handler'
        /usr/local/share/lua/5.1/kong/db/dao/plugins.lua:227: in function 'load_plugin'
        /usr/local/share/lua/5.1/kong/db/dao/plugins.lua:275: in function 'load_plugin_schemas'
        /usr/local/share/lua/5.1/kong/init.lua:425: in function 'init'
        init_by_lua:3: in main chunk
Error: /usr/local/share/lua/5.1/kong/cmd/start.lua:64: failed to start nginx (exit code: 1)

  Run with --v (verbose) or --vv (debug) for more details
```

In the code over here

https://github.com/Kong/go-pluginserver/blob/v0.3.0/pluginserver.go#L259

it fails. Looking at the code further over here which is part of golang
std library `plugin`

https://godoc.org/plugin#Plugin.Lookup

```
// Lookup searches for a symbol named symName in plugin p.
// A symbol is any exported variable or function.
// It reports an error if the symbol is not found.
// It is safe for concurrent use by multiple goroutines.
func (p *Plugin) Lookup(symName string) (Symbol, error) {
	return lookup(p, symName)
}
```

And symbol - https://godoc.org/plugin#Symbol

```
Symbol is a `pointer` to a `variable` or `function`.
```

You can see how they say it's a pointer. And there's also a sample code in
the same place where they show

```
*v.(*int) = 7
```

So, I changed the `go-pluginserver` code to this -

```
v, _ := plug.code.Lookup("Version")
if v != nil {
  info.Version = *v.(*string)
}

prio, _ := plug.code.Lookup("Priority")
if prio != nil {
  info.Priority = *prio.(*int)
}
```

And built it and installed it and tried it with kong and everything worked
and the log showed this

```
2020/01/15 14:19:09 [notice] 55391#0: *16 [kong] go.lua:69 Starting go-pluginserver, context: ngx.timer
```

I'll raise this as a PR. I'll make the code changes, but what about tests? I see there's a `test.sh`, but I couldn't understand it completely, like the `rq` command which I don't know. But I can see that the plugins in https://github.com/kong/go-plugins is being used to test, the `go-log` and `go-hello`, may be we could add `Version` and `Priority` fields in it to do the testing? Let me know 😄

--- a/pluginserver.go
+++ b/pluginserver.go
@@ -256,12 +256,12 @@ func (s *PluginServer) GetPluginInfo(name string, info *PluginInfo) error {
 
 	v, _ := plug.code.Lookup("Version")
 	if v != nil {
-		info.Version = v.(string)
+		info.Version = *v.(*string)
 	}
 
 	prio, _ := plug.code.Lookup("Priority")
 	if prio != nil {
-		info.Priority = prio.(int)
+		info.Priority = *prio.(*int)
 	}
 
 	// 	st, _ := getSchemaDict(reflect.TypeOf(plug.config).Elem())

